core/vm/evmc.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package vm
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/evmc/bindings/go/evmc"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type EVMC struct {
+	instance *evmc.Instance
+	env      *EVM
+	readOnly bool // TODO: The readOnly flag should not be here.
+}
+
+var (
+	createMu     sync.Mutex
+	evmcConfig   string // The configuration the instance was created with.
+	evmcInstance *evmc.Instance
+)
+
+func createVM(config string) *evmc.Instance {
+	createMu.Lock()
+	defer createMu.Unlock()
+
+	if evmcInstance == nil {
+		options := strings.Split(config, ",")
```

Comment:
I was considering `;` first, but `;` is a command separator in shell so you would need to enclose the options in `""`.
The `,` seems to be the best, because geth uses `,` for options accepting lists.